torchio/data/subject.py

```diff
@@ -71,7 +71,7 @@ def __copy__(self):
             else:
                 value = copy.deepcopy(value)
             result_dict[key] = value
-        new = Subject(result_dict)
+        new = self.__class__(**result_dict)
```

Comment:
```suggestion
        new = type(self)(**result_dict)
```
is also a bit cleaner :)